### ISSUE TITLE
Olive: list of new MFEs

### ISSUE THREAD

**Issue by pdpinch** [openedx/build-test-release-wg#200]
Having a list of new MFEs will help with planning, testing and documentation of olive. 

1. this list is still in progress
4. [frontend-app-course-authoring](https://github.com/openedx/frontend-app-course-authoring)
5. [frontend-app-authn](https://github.com/openedx/frontend-app-authn)
6. [frontend-app-learner-record](https://github.com/edx/frontend-app-learner-record/)

To be _included_ in the olive release, a MFE must:

1. have an olive branch
2. be added to the [tutor MFE plugin](https://github.com/overhangio/tutor-mfe) 

To be _supported_ in olive, a MFE must:

1. have sufficient documentation for a site operator to turn it on 
2. meet the criteria of https://github.com/openedx/build-test-release-wg/issues/131
3. be tested as part of the testing plan

**Comment by pdpinch** [comment]
In the nutmeg release notes, these MFEs were identified as "pre-alpha" and presumably should be considered for olive:

- [Authentication Micro-Frontend (MFE)](https://github.com/openedx/frontend-app-authn)
- [Course Authoring Micro-frontend (MFE)](https://github.com/openedx/frontend-app-course-authoring)
- [Discussions Micro-Frontend (MFE)](https://github.com/openedx/frontend-app-discussions)
- [Open-Response Assessments (ORA) Grading Micro-frontend (MFE)](https://github.com/edx/frontend-app-ora-grading)

**Comment by pdpinch** [comment]
The [Discussions MFE has `maybe: True` for inclusion in the open edX release](https://github.com/openedx/frontend-app-discussions/blob/master/openedx.yaml#L11). Unless that changes, it won't get a branch for olive.

**Comment by pdpinch** [comment]
I discovered another MFE, via its setting, [ORA_GRADING_MICROFRONTEND_URL](https://edx.readthedocs.io/projects/edx-platform-technical/en/latest/settings.html#setting-ORA_GRADING_MICROFRONTEND_URL) 

- [ORA grading microfrontend](https://github.com/edx/frontend-app-ora-grading)

**Comment by hurtstotouchfire** [comment]
@Tj-Tracy FYI

**Comment by pdpinch** [comment]
According to @nedbat in [Slack](https://openedx.slack.com/archives/C01AGTSB1LL/p1665168470278519?thread_ts=1665166971.526769&cid=C01AGTSB1LL), these are the MFEs that are branched: 

```
edx/frontend-app-communications
edx/frontend-app-learner-dashboard
edx/frontend-app-ora-grading
openedx/frontend-app-account
openedx/frontend-app-authn
openedx/frontend-app-course-authoring
openedx/frontend-app-ecommerce
openedx/frontend-app-gradebook
openedx/frontend-app-learner-record
openedx/frontend-app-learning
openedx/frontend-app-payment
openedx/frontend-app-profile
openedx/frontend-app-publisher
openedx/frontend-app-support-tools
openedx/frontend-template-application
```

**Comment by arbrandes** [comment]
Are we any closer to agreeing which ones will be added to the tutor-mfe plugin?

**Comment by arbrandes** [comment]
There's a PR to add `frontend-app-learner-record`, for the record: https://github.com/overhangio/tutor-mfe/pull/63

**Comment by regisb** [comment]
Considering the huge list of new MFEs that are expected to land in Olive, I'm very anxious that we merge the [runtime configuration feature](https://github.com/openedx/frontend-wg/issues/103). If we don't, I'm afraid we won't be able to release Olive at all. I suggest we have this conversation here: https://discuss.openedx.org/t/mfe-runtime-configuration/8495

**Comment by pdpinch** [comment]
I think we can safely omit [frontend-template-application](https://github.com/openedx/frontend-template-application/blob/master/openedx.yaml#L5) from the list of potential MFEs. I don't even know why it has an olive branch.

**Comment by arbrandes** [comment]
Copy-pasta of [a post](https://discuss.openedx.org/t/mfe-runtime-configuration/8495/7?u=arbrandes) I made in the forum:

#### Do we need to include every MFE in that list in Tutor/Olive? 

*TLDR: No.*

MFE developers (as of now mostly employed at or by 2U) are aware they can't remove a feature in edx-platform before the MFE that replaces it is officially accepted as being production-ready by the community.  This should apply to each MFE in that list, and it means we don't *have* to enable any of them for Olive if we don't want to.

#### How do we decide which ones to include?

*TLDR: Don't know.  We need to decide how to decide.*

While the ultimate decision on whether an MFE is ready is in fact up to the community, and in particular, the BTR group, there is no official method to arrive at this decision.  This is a problem we should fix *pronto*, precisely so we can answer the question of which ones to include.

[This comment](https://github.com/openedx/build-test-release-wg/issues/131#issuecomment-1270548031) in the ticket where we're tracking this question is a collation of the information I could find.  Please refer to it and add your comments, including which MFEs you would like to have included, and why.  Once we have a reasonable idea of where the community's opinion lies, we can start to narrow down the list in a technical fashion: by checking each MFE against the criteria.

(And yes, right off the bat it looks like a *sine qua non* criterion is whether the MFE supports runtime configuration.)

#### Product considerations

*TLDR: Let's consult the Product Working Group as well.*

There are product considerations to take into account.  Ultimately, I think the list will be narrowed down a lot by them.  I'm not saying they have ultimate say, but it doesn't make sense to start maintaining Tutor support for an MFE if it doesn't make sense to make it part of the Open edX product at a given point in time.

I've started discussing this with @jmakowski, yesterday.  The short of it is that my job will be to try and put all stakeholders together (mostly asynchronously, such as via this post) so we can decide ASAP which MFEs to work on for Olive.

**Comment by jmakowski1123** [comment]
@arbrandes @pdpinch Just noting the parallel ticket I'm working on here to try and get as much product-oriented data as possible for these. Each MFE has a ticket now on the community roadmap and a base level of acceptance criteria data: https://github.com/openedx/tcril-engineering/issues/495

**Comment by hurtstotouchfire** [comment]
FYI @arbrandes / @pdpinch I've worked with Jenna to get https://github.com/openedx/frontend-app-learner-record/issues/107 fleshed out for LRMFE. Our goal is for it to be included in Olive, off by default, for any adventurous beta adopters who have worked out a way to run the Credentials IDA. For Palm, we will aim to have [Credentials supported in Tutor](https://github.com/openedx/credentials/issues/1761) and [the old UI will be removed](https://github.com/openedx/credentials/issues/1722), so the MFE will be mandatory in Palm.

Note that https://github.com/overhangio/tutor-mfe/pull/63 is blocked pending a consensus on the merge target and documentation.

**Comment by arbrandes** [comment]
The following MFEs have at least one champion in the BTR group for inclusion/enablement by default in the `tutor-mfe` plugin for Olive, so will be prioritized in testing:

* frontend-app-course-authoring
* frontend-app-ora-grading

The following has an open PR for inclusion and no objections were raised, so it will be tested:

* frontend-app-learner-record

The following met no objections to "optional enablement", where there will be instruction in the Release Notes on how to get them running with Tutor, but they won't actually be part of the `tutor-mfe` plugin this release:

* frontend-app-authn
* frontend-app-support-tools

Last but not least, the group thinks frontend-app-discussions will likely not meet requirements in time, but a suggestion was made to make it its own plugin.  The question was raised: are the authors of frontend-app-discussions willing to create and maintain such a plugin?

**Comment by arbrandes** [comment]
It was raised [in the forum](https://discuss.openedx.org/t/whats-the-deal-with-tpa-hinted-links-in-nutmeg/8436/23?u=arbrandes) that frontend-app-authn has a (possibly) pressing reason to be included now, so I'm adding it to the championed list.

**Comment by arbrandes** [comment]
The final list of nominees:

* frontend-app-course-authoring
* frontend-app-discussion
* frontend-app-ora-grading
* frontend-app-learner-record
* frontend-app-authn

Their status against the [list of requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3561521275/Requirements+for+an+MFE) is also being kept in the wiki:

https://openedx.atlassian.net/wiki/spaces/COMM/pages/3571056752/Olive+MFE+Candidate+Requirements+Matrix

(There are some more details in comments [to this other issue](https://github.com/openedx/tcril-engineering/issues/494).)

Hopefully we'll have a decision on which ones we'll actually include [tomorrow](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3571908609/2022-11-07+BTR+Meeting+notes#%5CuD83D%5CuDDE3-Discussion-topics).

**Comment by arbrandes** [comment]
From yesterday's meeting, BTR have elected the following two new MFEs for potential inclusion, pending volunteers to fix individual issues.  If individual issues listed below are not fixed in the next two weeks (leaving two further weeks for testing/translation), they will not be included:

* frontend-app-course-authoring: 
  * flag to hide proctoring exam settings (if not already there on the side of edx-platform)
  * feature flag to hide discussions configuration
  * runtime configuration
  * patch to tutor-mfe

* frontend-app-ora-grading
  * runtime configuration
  * patch to tutor-mfe
